src/merkle/merkle_tree.rs

```diff
@@ -154,6 +161,63 @@ impl MerkleTree {
     }
 }
 
+/// Converts a [Word] into hex.
+fn word_to_hex(w: &Word) -> Result<String, fmt::Error> {
+    let mut s = String::new();
+
+    for byte in w.iter().flat_map(|e| e.to_bytes()) {
+        write!(s, "{byte:02x}")?;
+    }
+
+    Ok(s)
+}
```

Comment:
We already have a `to_hex()` utility in `miden-core` - should we use it here?

Reply:
I thought about it, this code is actually mostly based on that. The issue is that we would have a cyclic dependency, an alternative is to make miden-core use this instead, wdyt?

Reply:
Ah - makes sense! I forgot that `miden-core` depends on `miden-crypto` and not the other way around.

I think it's fine to keep it in two places for now - but I wonder if we should move this one into a dedicated utils modules (e.g., maybe under `src/utils.rs`).

Reply:
moved